File: api/app/schemas/comment_bank.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

CommentUsageStatus = Literal["unused", "used"]


class CommentBankItemRead(BaseModel):
    id: int
    doctor_id: int = Field(serialization_alias="doctorId")
    doctor_name: str = Field(serialization_alias="doctorName")
    keyword_id: int | None = Field(default=None, serialization_alias="keywordId")
    keyword: str = ""
    content: str
    status: CommentUsageStatus
    used_device_name: str | None = Field(default=None, serialization_alias="usedDeviceName")
    used_account: str | None = Field(default=None, serialization_alias="usedAccount")
    used_at: datetime | None = Field(default=None, serialization_alias="usedAt")
    created_at: datetime = Field(serialization_alias="createdAt")


class CommentBankImportResponse(BaseModel):
    imported: int
    skipped: int


class CommentBankBatchDeletePayload(BaseModel):
    ids: list[int] = Field(min_length=1)


class CommentBankBatchDeleteResponse(BaseModel):
    deleted: int

File: web_admin/src/types/commentBank.ts
export type CommentUsageStatus = 'unused' | 'used'

export type CommentBankItem = {
  id: number
  doctorId: number
  doctorName: string
  keywordId?: number | null
  keyword: string
  content: string
  status: CommentUsageStatus
  usedDeviceName?: string
  usedAccount?: string
  usedAt?: string
  createdAt: string
}

export type CommentBankQuery = {
  doctorId?: number | ''
  keywordId?: number | ''
  status?: CommentUsageStatus | ''
  keyword?: string
  page: number
  pageSize: number
}

export type CommentBankListResponse = {
  items: CommentBankItem[]
  total: number
}

export type CommentBankImportPayload = {
  doctorId: number
  file: File
}

export type CommentBankImportResponse = {
  imported: number
  skipped: number
}

export type CommentBankBatchDeletePayload = {
  ids: number[]
}

export type CommentBankBatchDeleteResponse = {
  deleted: number
}
